File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/ports/Atomic2SIn2AIn1Out.java
package tasks.ports;


import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Text;
@JavaTaskInfo
public class Atomic2SIn2AIn1Out extends JavaTask {

	@InputPortInfo(name = "In1", dataType = Text.class)
	public InputPort in1;
	@InputPortInfo(name = "In2", dataType = Text.class)
	public InputPort in2;
	@InputPortInfo(name = "In3", dataType = Text.class, asynchronous = true)
	public InputPort in3;
	@InputPortInfo(name = "In4", dataType = Text.class, asynchronous = true)
	public InputPort in4;
	@OutputPortInfo(name = "Out", dataType = Text.class)
	public OutputPort out;

	@Override
	public void executeAsync(InputPort asynchIn) {
		Text asynchInText = asynchIn.getData(Text.class);
		out.putData(new Text("From port " + asynchIn.getName() + ": " + asynchInText.getValue()));
	}

	@Override
	public void execute(int grp) {
		Text in1Text = in1.getData(Text.class);
		Text in2Text = in2.getData(Text.class);
		out.putData(new Text(in1Text.getValue() + " " + in2Text.getValue()));
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/ports/Atomic2SIn2AIn1Out.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Text import Text

@PythonTaskInfo
class Atomic2SIn2AIn1Out(PythonTask):

    def __init_ports__(self):
        self.in1 = InputPort(name="in1", data_type=Text)
        self.in2 = InputPort(name="in2", data_type=Text)
        self.in3 = InputPort(name="in3", data_type=Text, asynchronous=True)
        self.in4 = InputPort(name="in4", data_type=Text, asynchronous=True)
        self.out = OutputPort(name="out", data_type=Text)

    def execute_async(self, async_in):
        asynchInText = async_in.get_data(Text)
        self.out.put_data(Text("From port " + async_in.name + ": " + asynchInText.values[0]))

    def execute(self, grp):
        in1Text = self.in1.get_data(Text)
        in2Text = self.in2.get_data(Text)
        self.out.put_data(Text(in1Text.values[0] + " " + in2Text.values[0]))

